### PR TITLE
Add some infinite and reversed perspectives.

### LINKED ISSUE #573: Projection: implement reversed-Z and infinite zFar
Reversed-Z and infinite zFar are two techniques that improves depth precision, mitigating issues like Z-fighting. The implementation is described [here](http://dev.theomader.com/depth-precision/).

These approaches are gradually gaining popularity as default configuration in game engines. Especially, infinite zFar makes things simpler and should be the go-to configuration to be used.

Reversed-Z has a different Z representation: thus it's backward incompatible and needs a different API. Infinite zFar should be implemented as an addition to the API.

### PR DESCRIPTION
**Pull request by sebcrozet**
This adds the following to nalgebra-glm:

- infinite_perspective_rh_no
- infinite_perspective_rh_zo
- ~reversed_perspective_rh_no~
- reversed_perspective_rh_zo
- ~reversed_infinite_perspective_rh_no~
- reversed_infinite_perspective_rh_zo

Addresses #573

### CODE REVIEW

**Comment by Ralith** [comment]
Reviewing [the article](http://dev.theomader.com/depth-precision/), it looks like I misremembered: reversed Z actually works fine on integer depth buffers, it's just that *forward* Z is terrible on *floating-point* depth buffers.

**Comment by ishitatsuyuki** [comment]
You should *not* use reversed Z with OpenGL ([-1, 1]) range.

> This doesn't make a difference for integer formats, but with floating-point, all the precision is stuck uselessly in the middle. (The value gets mapped into [0, 1] for storage in the depth buffer later, but that doesn't help, since the initial mapping to [-1, 1] has already destroyed all the precision in the far half of the range.) And by symmetry, the reversed-Z trick will not do anything here. 
> *[Depth Precision Visualized - NVIDIA Developer](https://developer.nvidia.com/content/depth-precision-visualized)*

Also, do you mind adding functions for the `nalgebra` types too? It doesn't have to be the same PR, but in that case I think you should leave the issue open.

File: nalgebra-glm/src/ext/matrix_clip_space.rs [hunk 1]
```diff
@@ -691,10 +687,110 @@ pub fn perspective_zo<N: RealField>(aspect: N, fovy: N, near: N, far: N) -> TMat
     perspective_rh_zo(aspect, fovy, near, far)
 }
 
+/// Build infinite right-handed perspective projection matrix with [-1,1] depth range.
+///
+/// # Parameters
+///
+/// * `fovy` - Field of view, in radians
+/// * `aspect` - Ratio of viewport width to height (width/height)
+/// * `near` - Distance from the viewer to the near clipping plane.
```
Comment (Ralith) [inline nalgebra-glm/src/ext/matrix_clip_space.rs]: Could these (and similar below) be listed in the order they're passed to the function?

**Comment by Ralith** [comment]
Would it make sense for the documentation to specifically encourage people to use `reversed_infinite_perspective_rh_zo`? I think it's pretty clearly the sane default option, so long as you're able to set up your depth range and depth test appropriately.

**Comment by sebcrozet** [comment]
Thank you @ishitatsuyuki and @Ralith for your reviews!
@Ralith I've not modified the documentation to encourage the use of reversed perspective as this has implications on the depth-range and depth test settings that may not be what you find on common tutorials. We could add this kind of advice to the user guide though.

**Pull request merged**